pkg/placement/placement.go

```diff
@@ -53,24 +52,35 @@ const (
 //  3. for updates on label selectors, re-evaluate if existing objects should be removed
 //     from clusters.
 func (c *Controller) handlePlacement(obj runtime.Object) error {
-	placement := obj.DeepCopyObject()
+	placement, err := runtimeObjectToPlacement(obj.DeepCopyObject())
+	if err != nil {
+		return err
+	}
 
 	// handle requeing for changes in placement, excluding deletion
 	if !isBeingDeleted(obj) {
+		// update placement resolution destinations since placement was updated
+		list, err := ocm.ListClustersBySelectors(c.ocmClient, placement.Spec.ClusterSelectors)
```

Comment:
Every caller of `ocm.ListClustersBySelectors` would be better off if that func returned a Set of strings rather than a slice of strings.